ml-proto/src/arithmetic.ml

```diff
@@ -119,10 +123,36 @@ struct
 
   let unop op =
     let f = match op with
-      | Clz -> fun i -> i  (* TODO *)
-      | Ctz -> fun i -> i  (* TODO *)
-      | Popcnt -> fun i -> i  (* TODO *)
-    in fun v -> Int.to_value (f (Int.of_value 1 v))
+      | Clz -> fun (v : Int.t) -> 
+                 let rec helper n acc =
+                   match n with 
+                     | n when n = Int.zero -> Int.size 
```

Comment:
Yeah, I'd like that, but unfortunately, Int.zero is a value and not a constructor so it can't be used in the pattern match directly, only as a guard. In my earlier iteration, I was thinking of dispensing with the match altogether but it does seem to be clearer in the later cases.